gotham/src/helpers/http/response/mod.rs

```diff
@@ -442,12 +465,20 @@ pub fn set_headers(state: &State, res: &mut Response, mime: Option<Mime>, length
 /// # assert!(response.headers().get::<XRequestId>().is_some());
 /// # }
 /// ```
-pub fn set_redirect_headers<L: Into<Cow<'static, str>>>(
+pub fn set_redirect_headers<B, L: Into<Cow<'static, str>>>(
     state: &State,
-    res: &mut Response,
+    res: &mut Response<B>,
     location: L,
 ) {
     let headers = res.headers_mut();
-    headers.set(XRequestId(request_id(state).into()));
-    headers.set(Location::new(location));
+    set_request_id(state, headers);
+    headers.insert(LOCATION, location.into().to_string().parse().unwrap());
+}
+
+/// Sets the request id inside a given `HeaderMap`.
+fn set_request_id(state: &State, headers: &mut HeaderMap) {
```

Comment:
Maybe we can resurrect the `headers` module with default header functions like this for `set/get`. We can use statics for the `HeaderName` then to avoid converting often.